### PR TITLE
[CPU] Coverity fixes

### PR DESCRIPTION
**Pull request by egorshul**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *79251*
 - *79260*
 - *79234*
 - *79242*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@maxnick please review the changes

File: src/plugins/intel_cpu/src/mkldnn_exec_network.cpp [hunk 1]
```diff
@@ -503,10 +505,12 @@ bool MKLDNNExecNetwork::CanProcessDynBatch(const InferenceEngine::CNNNetwork &ne
         auto type = TypeFromName(op->get_type_name());
         if (type == Tile) {
             const auto tile = std::dynamic_pointer_cast<const ngraph::opset1::Tile>(op);
+            if (!tile)
+                return false;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/mkldnn_exec_network.cpp:509]: The previous logic called `continue` if the `tile` is null, which probably means that only `opset1::Tile` node supported, but here we interrupt the whole loop. Should not we align the behavior with the previous logic? BTW `get_input_node_shared_ptr` can be called to the base node type.

File: src/plugins/intel_cpu/src/mkldnn_exec_network.cpp [hunk 2]
```diff
@@ -423,11 +423,9 @@ bool MKLDNNExecNetwork::canBeExecViaLegacyDynBatch(std::shared_ptr<const ov::Mod
 
         if (type == Tile) {
             const auto tile = std::dynamic_pointer_cast<const ngraph::opset1::Tile>(op);
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/mkldnn_exec_network.cpp:425]: Could you please move `tile` initialization closer to its usage in the `if` statement?

File: src/plugins/intel_cpu/src/mkldnn_exec_network.cpp [hunk 3]
```diff
@@ -505,12 +503,10 @@ bool MKLDNNExecNetwork::CanProcessDynBatch(const InferenceEngine::CNNNetwork &ne
         auto type = TypeFromName(op->get_type_name());
         if (type == Tile) {
             const auto tile = std::dynamic_pointer_cast<const ngraph::opset1::Tile>(op);
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/mkldnn_exec_network.cpp:505]: Could you please move `tile` initialization closer to its usage in the `if` statement?

**Review by maxnick — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**